forest/src/daemon.rs

```diff
@@ -37,11 +37,15 @@ pub(super) async fn start(config: Config) {
             Keypair::Ed25519(gen_keypair)
         });
 
+    // Initialize keystore
+    let keystore = Arc::new(RwLock::new(PersistentKeyStore::new(
+        config.data_dir.to_string(),
```

Comment:
uh, this is just going to try to open the directory as a file, you should change this or the logic inside the keystore to suffix the path to the directory/file it will be placed in